### PR TITLE
Airlock construction and deconstruction

### PR DESCRIPTION
**Pull request by Ahion**
### Purpose
Part of #5819. Add Airlock construction and deconstruction. To deconstruction the airlock, it must be de-energized, welded and the bolts will be up. Made an interface for airlock electronics. Access can be configured. Details in the video:
https://youtu.be/GBoMUkT39Lw
### Notes:
I spent a lot of time trying to make a spoiler in the metal build menu. But everything that I got worked only on the server side. So I deleted it. I only added one airlock so as not to overload the metal building menu. When I add a spray painter, the airlock can be painted in the desired form.

I have not added naming the Airlock with a pen. I don't know if we have any input field that I could use for this?

### Changelog:

 CL: [New] Added airlock construction and deconstruction

### CODE REVIEW

File: UnityProject/Assets/Scripts/Objects/Doors/ConstructibleDoor.cs [hunk 1]
```diff
@@ -1,56 +1,120 @@
...
-			if (DefaultWillInteract.Default(interaction, side) == false) return false;
-			if (Validations.HasItemTrait(interaction.UsedObject, CommonTraits.Instance.Screwdriver) == false)
+			if (!DefaultWillInteract.Default(interaction, side) || interaction.TargetObject != gameObject)
 				return false;
-			return true;
+
+			if (Validations.HasItemTrait(interaction.UsedObject, CommonTraits.Instance.Screwdriver))
+				return true;
+
+			if (!weldModule.CanDoorStateChange() && boltsModule.CanDoorStateChange() && !doorMasterController.HasPower)
```
Comment (MaxIsJoe) [inline UnityProject/Assets/Scripts/Objects/Doors/ConstructibleDoor.cs]: `if (weldModule.CanDoorStateChange() == false && boltsModule.CanDoorStateChange() && doorMasterController.HasPower == false)`

File: UnityProject/Assets/Scripts/Objects/Doors/ConstructibleDoor.cs [hunk 2]
```diff
@@ -1,56 +1,120 @@
...
+						$"{interaction.Performer.ExpensiveName()} screws in {gameObject.ExpensiveName()}'s cable panel.");
+
+					//Force close net tab when panel is closed
+					TabUpdateMessage.SendToPeepers(gameObject, NetTabType.HackingPanel, TabAction.Close);
+				}
+
+
+				AudioSourceParameters audioSourceParameters =
+					new AudioSourceParameters(pitch: UnityEngine.Random.Range(0.8f, 1.2f));
+				SoundManager.PlayNetworkedAtPos(CommonSounds.Instance.screwdriver,
```
Comment (MaxIsJoe) [inline UnityProject/Assets/Scripts/Objects/Doors/ConstructibleDoor.cs:89]: The doc string recommends that you use the async method of this
don't forget to add `_ =` if you choose to use that method

**Comment by MaxIsJoe** [comment]
This is very pog

**Comment by corp-0** [comment]
to make the tests happy, run the tests locally and commit the changes the Networking test makes.

**Comment by NoooneyDude** [comment]
> I have not added naming the Airlock with a pen. I don't know if we have any input field that I could use for this?

I don't know much of it myself, but I think the hand labeler has some function?

File: UnityProject/Assets/Scripts/Objects/Doors/ConstructibleDoor.cs [hunk 3]
```diff
@@ -1,56 +1,120 @@
...
-				DoorAnimatorV2.AddPanelOverlay();
-				Chat.AddActionMsgToChat(interaction.Performer,
-					$"You unscrews the {gameObject.ExpensiveName()}'s cable panel.",
-					$"{interaction.Performer.ExpensiveName()} unscrews {gameObject.ExpensiveName()}'s cable panel.");
+				panelopen = !panelopen;
+				if (panelopen)
+				{
+					DoorAnimatorV2.AddPanelOverlay();
+					Chat.AddActionMsgToChat(interaction.Performer,
+						$"You unscrews the {gameObject.ExpensiveName()}'s cable panel.",
```
Comment (NoooneyDude) [inline UnityProject/Assets/Scripts/Objects/Doors/ConstructibleDoor.cs]:
```suggestion
						$"You unscrew the {gameObject.ExpensiveName()}'s cable panel.",
```

**Review by NoooneyDude — COMMENTED** [review]
Thank you!

File: UnityProject/Assets/Scripts/Objects/Doors/ConstructibleDoor.cs (hunk 2 above)
Comment (Ahion) [reply UnityProject/Assets/Scripts/Objects/Doors/ConstructibleDoor.cs:89]:
```suggestion
				_ = SoundManager.PlayNetworkedAtPos(CommonSounds.Instance.screwdriver,
```
Do you mean that? But this method returns nothing. I don't quite understand why I need to use discard here. Actually, this code was already here, I did not write it

File: UnityProject/Assets/Scripts/Objects/Doors/ConstructibleDoor.cs (hunk 1 above)
Comment (Ahion) [reply UnityProject/Assets/Scripts/Objects/Doors/ConstructibleDoor.cs]: done

File: UnityProject/Assets/Scripts/Objects/Doors/ConstructibleDoor.cs (hunk 3 above)
Comment (Ahion) [reply UnityProject/Assets/Scripts/Objects/Doors/ConstructibleDoor.cs]: done

**Comment by Ahion** [comment]
> I don't know much of it myself, but I think the hand labeler has some function?

exactly! thanks for the hint

**Comment by PerfectTangent** [comment]
@Ahion if you could make the pen naming function generic, it would be useful for other applications.

**Comment by Ahion** [comment]
> if you could make the pen naming function generic, it would be useful for other applications.

How should it work? Make a pen method that asks the user for input and returns it as a string?

**Review by Bod9001 — APPROVED** [review]

File: UnityProject/Assets/Scripts/Objects/Doors/ConstructibleDoor.cs (hunk 2 above)
Comment (NoooneyDude) [reply UnityProject/Assets/Scripts/Objects/Doors/ConstructibleDoor.cs:89]: You can discard the result of the async method - I think it is `PlayNetworkedAtPosAsync`?

**Review by NoooneyDude — APPROVED** [review]

**Pull request merged**